qbraid/runtime/batch.py

```
@@ -0,0 +1,401 @@
+            timeout: Maximum seconds to wait. None = wait indefinitely.
+            poll_interval: Seconds between status checks per job.
+
+        Returns:
+            BatchResult mapping jobQrn -> Result[ResultDataType].
+
+        Raises:
+            TimeoutError: If timeout reached before all jobs complete.
+            RuntimeError: If batch session was never opened.
+        """
+        if self._batch_data is None:
+            raise RuntimeError("Batch session has not been opened.")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)